indico/modules/events/abstracts/forms.py

```diff
@@ -498,7 +498,9 @@ def __init__(self, *args, **kwargs):
         if not is_invited:
             self.person_links.require_speaker_author = abstracts_settings.get(self.event, 'speakers_required')
             self.person_links.allow_speakers = abstracts_settings.get(self.event, 'allow_speakers')
-            self.person_links.disable_user_search = session.user is None
+            self.person_links.disable_user_search = ((not config.ALLOW_PUBLIC_USER_SEARCH) and
+                                                     (not self.event.can_manage(session.user)) or
+                                                     (session.user is None))
```

Comment:
Not really what I meant in my last comment - you still have `and` and `or` on the same level.

What I want to avoid is `a and b or c` but rather being explicit using `(a and b) or c` / `a and (b or c)`